go/integTest/readfrom_strategy_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package integTest
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/valkey-io/valkey-glide/go/api"
+	"github.com/valkey-io/valkey-glide/go/api/config"
+	"github.com/valkey-io/valkey-glide/go/api/options"
+)
+
+func (suite *GlideTestSuite) TestRoutingWithAzAffinityStrategyTo1Replica() {
+	suite.SkipIfServerVersionLowerThanBy("8.0.0")
+	az := "us-east-1a"
+	const GET_CALLS = 3
+	getCmdStat := "cmdstat_get:calls=" + fmt.Sprint(GET_CALLS)
+
+	clientForConfigSet := suite.clusterClient(api.NewGlideClusterClientConfiguration().
+		WithAddress(&suite.clusterHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(2000))
+
+	// Reset the availability zone for all nodes
+	_, err := clientForConfigSet.CustomCommandWithRoute(
+		[]string{"CONFIG", "SET", "availability-zone", ""}, config.AllNodes)
+	assert.NoError(suite.T(), err)
+	resetStatResult, err := clientForConfigSet.CustomCommand([]string{"CONFIG", "RESETSTAT"})
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), "OK", resetStatResult.SingleValue())
+
+	// 12182 is the slot of "foo"
+	_, err = clientForConfigSet.CustomCommandWithRoute(
+		[]string{"CONFIG", "SET", "availability-zone", az}, config.NewSlotIdRoute(config.SlotTypeReplica, 12182))
+	assert.NoError(suite.T(), err)
+
+	clientForTestingAz := suite.clusterClient(api.NewGlideClusterClientConfiguration().
+		WithAddress(&suite.clusterHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(2000).
```

Comment:
I think you can introduce `suite.defaultClusterClientConfig()` and re-use it there (and in other tests) and inside `suite`